### PR TITLE
add implementation of method exec for Container

### PR DESCRIPTION
**Pull request by lumimul**
PR for #168  
Added method to create an ``Exec``.

### CODE REVIEW

**Comment by 0crat** [comment]
Election failed: performer was not elected (see footprint for details)

**Comment by 0crat** [comment]
Election failed: performer was not elected (see footprint for details)

**Comment by 0crat** [comment]
Election failed: performer was not elected (see footprint for details)

**Comment by 0crat** [comment]
Election failed: performer was not elected (see footprint for details)

**Comment by 0crat** [comment]
Election failed: performer was not elected (see footprint for details)

**Comment by 0crat** [comment]
Election failed: performer was not elected (see footprint for details)

**Comment by 0crat** [comment]
Election failed: performer was not elected (see footprint for details)

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) not enough funds available in the project, can't set budget of job [#337](https://github.com/amihaiemil/docker-java-api/issues/337), see [§21](http://www.zerocracy.com/policy.html#21); @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will get no money on completion; in order to fix that, add funds to the project and assign the job again

File: src/main/java/com/amihaiemil/docker/Execs.java [hunk 1]
```diff
@@ -34,6 +37,15 @@
  */
 public interface Execs {
 
+    /**
+     * Create a exec with ID or name of container.
+     * @param id ID or name of container.
+     * @param config Exec configuration.
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/Execs.java:43]: @lumimul Please also add the tag:

``@see <a href="https://docs.docker.com/engine/api/v1.40/#operation/ContainerExec">Create Exec</a>``

So the users know where to look for the Json Config format.

File: src/main/java/com/amihaiemil/docker/Execs.java [hunk 2]
```diff
@@ -34,6 +37,15 @@
...
+     * @return Created Exec.
+     * @throws IOException If something goes wrong.
+     */
+    Exec create(final String id, final JsonObject config) throws IOException;
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/Execs.java:47]: @lumimul This method should be on the ``Container`` interface and should only work if the Container is running. According to the [documentation](https://docs.docker.com/engine/api/v1.40/#operation/ContainerExec): "Run new commands inside running containers"

Also, the path of the request is ``/containers/{id}/exec``, which is another indicator that it should be on the ``Container`` interface: path building throughout the library should be fluent, we shouldn't have to replace strings or do other kind of manipulations.

Comment (lumimul) [reply src/main/java/com/amihaiemil/docker/Execs.java:47]: i am not understand what this method will return
id of exec or  RtExec? if RtExec need replace uri (baseuri/container/hash -> baseuri/exec/hash)

Comment (amihaiemil) [reply src/main/java/com/amihaiemil/docker/Execs.java:47]: @lumimul This method will return the created ``Exec``, like this:

```java
final class RtContainers implements Containers {
    //...
    private final Docker docker;

    //...

    public Exec create(final JsonObject config){
        //do the HTTP call
        final String execId = ...;
        return this.docker.execs().get(execId);
    } 
    //...
}
```

File: src/main/java/com/amihaiemil/docker/Execs.java (hunk 1 above)
Comment (lumimul) [reply src/main/java/com/amihaiemil/docker/Execs.java]: fix it

File: src/main/java/com/amihaiemil/docker/Execs.java (hunk 2 above)
Comment (lumimul) [reply src/main/java/com/amihaiemil/docker/Execs.java]: fix it

**Comment by amihaiemil** [comment]
@lumimul Looks good, thanks!

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by amihaiemil** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#337](https://github.com/amihaiemil/docker-java-api/issues/337) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't set title

**Comment by 0crat** [comment]
Job was finished in 44 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by amihaiemil** [comment]
@lumimul Thanks a lot!

**Pull request merged**